qiskit/opflow/__init__.py

```diff
@@ -61,6 +61,16 @@
 3. An n-qubit Measurement is a functional taking n-qubit State functions to complex values.
 For example, a Pauli Z Measurement can be defined by f(Zero) = 0 and f(One) = 1.
 
+.. note::
+
+    While every effort has been made to make programming the Operator Flow similar to mathematical
+    notation, in some places our hands are tied by the design of Python.  In particular, when using
+    mathematical operators such as ``+`` and ``^`` (tensor product), beware that these follow
+    `Python operator precedence rules
+    <https://docs.python.org/3/reference/expressions.html#operator-precedence>`__.  For example,
+    ``I^X + X^I`` will actually be interpreted as ``I ^ (X+X) ^ I == 2 * I^X^I``.  In these cases,
+    you should use extra parentheses, or use the relevant method calls.
```

Comment:
It might be nice to add the parenthesis example too just to be clear about where users should put them (should be obvious but an example never hurts) 😄 
```suggestion
    ``I^X + X^I`` will actually be interpreted as ``I ^ (X+X) ^ I == 2 * I^X^I``.  In these cases,
    you should use extra parentheses, like ``(I ^ X) + (X ^ I)``, or use the relevant method calls.
```

Reply:
Sounds good!  Done in 60cc807.